sample/rules_test.go

```diff
@@ -1060,3 +1060,463 @@ func TestRuleMatchesSpanMatchingSpan(t *testing.T) {
 		})
 	}
 }
+
+func TestRulesDatatypes(t *testing.T) {
```

Comment:
We basically need an entirely new set of tests that includes Datatype in the conditions, that exercise converting the different input types to the appropriate datatype. 

I suggest writing a couple of helper functions to build Conditions and Spans.

Reply:
Sorry, I added back the datatype information. Should resolve this.